client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -243,13 +253,29 @@ const PatternAssembler: Step = ( { navigation, flow, stepName } ) => {
 		updateActivePatternPosition( position - 1 );
 	};
 
-	const onSelect = ( type: string, selectedPattern: Pattern | null ) => {
-		if ( type ) {
+	const onSelect = (
+		type: string,
+		selectedPattern: Pattern | null,
+		categorySelected: string | null
+	) => {
+		if ( selectedPattern ) {
+			// Inject the selected pattern category or the first category
+			// because it's used in tracks and as pattern name in the list
+			selectedPattern.category = categories.find( ( { name } ) => name === categorySelected );
+
 			trackEventPatternSelect( {
-				patternType: type,
-				patternId: selectedPattern?.id || 0,
-				patternName: selectedPattern?.name || '',
+				patternType: selectedPattern.category?.name || type,
+				patternId: selectedPattern.id,
```

Comment:
We use the `null` value to track deselecting header and footer. If we wrap the event inside the check, we're not able to know whether the user removes the header and footer. Any thougths?

Reply:
In tracks, other ways exist to know when the header and footer are not selected. In `calypso_signup_pattern_assembler_continue_click` with the prop `pattern_types` or in the event `calypso_signup_pattern_assembler_pattern_final_select` with the prop `pattern_category`.


I think `calypso_signup_pattern_assembler_pattern_select_click` makes more sense to track patterns selected rather than patterns not selected.

Reply:
Okay, that also makes sense to me 🙂